OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java

```diff
@@ -454,8 +454,30 @@ private Set<String> getLanguages(@NonNull Amenity amenity) {
 		return langs;
 	}
 
-	private void sortSearchResults(@NonNull List<WikivoyageSearchResult> results) {
-		Collections.sort(results, (o1, o2) -> collator.compare(o1.getArticleTitle(), o2.getArticleTitle()));
+	public void sortSearchResults(@NonNull List<WikivoyageSearchResult> results, String searchQuery) {
+		String searchQueryLC = searchQuery.toLowerCase();
+		results.sort((sr1, sr2) -> {
+			int sr1Comparison = collator.compare(sr1.getArticleTitle(), searchQuery);
+			int sr2Comparison = collator.compare(sr2.getArticleTitle(), searchQuery);
+
+			if (sr1Comparison == 0 && sr2Comparison != 0) {
+				return -1;
```

Comment:
comparator should always return Integer.compare or String.compare methods to avoid in consistency, this comparator should be rewritten without ifs  !=, check Search results comparator